output/bootstrap.go

```diff
@@ -4,20 +4,37 @@
 
 package output
 
-import (
-	"fmt"
-)
+import "fmt"
 
 // GenerateBootstraps : generates necessary bootstraps for instances
-func GenerateBootstraps(instances []Instance) (bootstraps []Execution) {
+func GenerateBootstraps(m *FSMMessage) []Execution {
+	var bootstraps []Execution
+	var endpoint string
+	var execType string
+
+	if m.ServiceIP != "" {
+		endpoint = m.ServiceIP
+	} else {
+		endpoint = "$(routers.items.0.ip)"
+	}
+
+	if m.Datacenters.Items[0].Type == "fake" {
```

Comment:
Same as avobe, fake datacenters can be "vcloud-fake" and "aws-fake".
May be good to centralize that logic to datacenter struct so you can do something like:
```
if m.Datacenters.Items[0].IsFake {
}
```